src/test/java/jp/co/soramitsu/sora/service/DIDServiceImplTest.java

```diff
@@ -0,0 +1,96 @@
+package jp.co.soramitsu.sora.service;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+
+import com.vladmihalcea.hibernate.type.json.internal.JacksonUtil;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import jp.co.soramitsu.sora.didresolver.domain.iroha.repositories.AccountRepository;
+import jp.co.soramitsu.sora.didresolver.domain.iroha.valueobjects.DID;
+import jp.co.soramitsu.sora.didresolver.dto.DDO;
+import jp.co.soramitsu.sora.didresolver.exceptions.UnparseableException;
+import jp.co.soramitsu.sora.didresolver.services.StorageService;
+import jp.co.soramitsu.sora.didresolver.services.impl.DIDServiceViaRepository;
+import lombok.val;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.theories.DataPoints;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+@RunWith(Theories.class)
+public class DIDServiceImplTest {
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @DataPoints
+  public static List<String> incorrectDids = new ArrayList<>();
+
+  @Mock
+  private AccountRepository accountRepository;
+
+  private StorageService storageService;
+
+  @Before
+  public void setUp() {
+    String[] invalidDidsStringArr = {
+        "plainly incorrect did",
```

Comment:
I suggest to take out a set of incorrect DIDs in base class or source provider or file